pyvo/mivot/utils/mivot_utils.py

```diff
@@ -118,3 +193,127 @@ def cast_type_value(value, dmtype):
             return float(value)
         else:
             return value
+
+    @staticmethod
+    def format_dmid(dmid):
+        """
+        Replace characters that could confuse XPath queries with '_'.
+        This is not required by the MIVOT schema but this makes this API more robust
+
+        Returns
+        -------
+        str
+            formatted dmid
+        """
+        if dmid is not None:
+            return dmid.replace("/", "_").replace(".", "_").replace("-", "_")
+        return ""
+
+    @staticmethod
+    def get_field_attributes(table, column_id):
+        """
+        Parameters
+        ----------
+        table : astropy.table
+            Table (from parsed VOTable) of the mapped data
+        column_id : str
+            Identifier of the table column from which we want to get the unit
+
+        Returns
+        -------
+        unit, ref, literal
+        """
+        ref, literal = MivotUtils.get_ref_or_literal(column_id)
+        if literal:
+            return None, None, literal
+        else:
+            try:
+                field = table.get_field_by_id_or_name(ref)
+                return str(field.unit), column_id, None
+            except KeyError as keyerror:
+                raise MappingError() from keyerror
```

Comment:
shouldn't this MappingError have some error message? afaik the text raised in the original KeyError is not propagated, so this will have no message at the end.